libs/designsystem/src/lib/components/calendar/calendar.component.ts

```diff
@@ -336,17 +337,19 @@ export class CalendarComponent implements OnInit, AfterViewInit, OnChanges {
         .clone()
         .date(newDay.date)
         .toDate();
-      this.onSelectedDateChange(selectedDate);
-      this._selectedDate = selectedDate;
 
-      if (this.timezone === 'local') {
-        // emit as local time midnight
-        this.dateChange.emit(selectedDate);
-      } else {
-        // emit as equivalent date in utc midnight
-        const utcMidnightDate = moment.utc(moment(selectedDate).format('YYYY-MM-DD')).toDate();
-        this.dateChange.emit(utcMidnightDate);
+      // emit equivalent date in utc midnight if timezone is not local
+      const emitDate =
+        this.timezone === 'local'
+          ? selectedDate
+          : moment.utc(moment(selectedDate).format('YYYY-MM-DD')).toDate();
+
+      if (!this._selectedDate || selectedDate.getTime() !== this._selectedDate.getTime()) {
```

Comment:
Consider using `hasDateChanged` helper method (also handles null checks).
Also consider renaming local variable to avoid confusion: `const selectedDate` => `const newDate` or `const newSelectedDate` or similar...
```suggestion
      if (this.hasDateChanged(selectedDate, this.selectedDate)) {
```

Reply:
Should be resolved with new commits.